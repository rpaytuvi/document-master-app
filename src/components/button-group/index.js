import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View} from 'react-native';
import {GhostButton} from '../';
import Styles from './styles';

class ButtonGroup extends Component {
  static propTypes = {
    textLeft: PropTypes.string,
    iconLeft: PropTypes.string,
    textRight: PropTypes.string,
    iconRight: PropTypes.string,
    onPressLeft: PropTypes.func.isRequired,
    onPressRight: PropTypes.func.isRequired,
    leftBigger: PropTypes.bool,
    selectedIndex: PropTypes.number,
  };

  static defaultProps = {
    textLeft: '',
    iconLeft: '',
    textRight: PropTypes.string,
    iconRight: PropTypes.string,
    onPressLeft: () => {},
    onPressRight: () => {},
    leftBigger: false,
    selectedIndex: -1,
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <View
          style={
            this.props.leftBigger
              ? Styles.buttonContainerBig
              : Styles.buttonContainer
          }>
          <GhostButton
            text={this.props.textLeft}
            icon={this.props.iconLeft}
            onPress={this.props.onPressLeft}
            customStyle={Object.assign(
              {},
              Styles.leftButton,
              this.props.selectedIndex > -1 ? Styles.buttonUnselected : {},
              this.props.selectedIndex === 0 ? Styles.buttonSelected : {},
            )}
            customTextStyle={Styles.buttonText}
            customIconStyle={Object.assign(
              {},
              Styles.buttonIcon,
              this.props.selectedIndex === 0 ? Styles.buttonIconSelected : {},
            )}
          />
        </View>
        <View style={Styles.buttonContainer}>
          <GhostButton
            text={this.props.textRight}
            icon={this.props.iconRight}
            onPress={this.props.onPressRight}
            customStyle={Object.assign(
              {},
              Styles.rightButton,
              this.props.selectedIndex > -1 ? Styles.buttonUnselected : {},
              this.props.selectedIndex === 1 ? Styles.buttonSelected : {},
            )}
            customTextStyle={Styles.buttonText}
            customIconStyle={Object.assign(
              {},
              Styles.buttonIcon,
              this.props.selectedIndex === 1 ? Styles.buttonIconSelected : {},
            )}
          />
        </View>
      </View>
    );
  };
}

export default ButtonGroup;
