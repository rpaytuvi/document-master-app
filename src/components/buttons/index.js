import GhostButton from './ghost';
import HeaderButton from './header';
import MainButton from './main';

export {GhostButton, HeaderButton, MainButton};
