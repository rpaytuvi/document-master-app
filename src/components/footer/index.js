import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {View} from 'react-native';
import {MainButton} from '../';
import Styles from './styles';

class Footer extends Component {
  static propTypes = {
    buttonText: PropTypes.string.isRequired,
    buttonIcon: PropTypes.string,
    action: PropTypes.func.isRequired,
  };

  static defaultProps = {
    buttonText: '',
    buttonIcon: '',
    action: () => {},
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <MainButton
          text={this.props.buttonText}
          icon={this.props.buttonIcon}
          onPress={this.props.action}
        />
      </View>
    );
  };
}

export default Footer;
