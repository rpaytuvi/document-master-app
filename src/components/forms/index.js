import DocumentForm from './document';

export {DocumentForm};
