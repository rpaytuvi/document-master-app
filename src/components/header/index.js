import React, {Component} from 'react';
import PropTypes from 'prop-types';
import {Text, View} from 'react-native';
import {HeaderButton} from '../';
import Styles from './styles';

class Header extends Component {
  static propTypes = {
    title: PropTypes.string.isRequired,
    action: PropTypes.func.isRequired,
    withAlert: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    title: '',
    action: () => {},
    withAlert: false,
  };

  render = () => {
    return (
      <View style={Styles.container}>
        <View style={Styles.titleContainer}>
          <Text style={Styles.title}>{this.props.title}</Text>
        </View>
        <View style={Styles.buttonContainer}>
          <HeaderButton
            onPress={this.props.action}
            withAlert={this.props.withAlert}
          />
        </View>
      </View>
    );
  };
}

export default Header;
