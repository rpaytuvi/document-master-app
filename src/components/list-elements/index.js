import DocumentLargeListElement from './document-large';
import DocumentSmallListElement from './document-small';

export {DocumentLargeListElement, DocumentSmallListElement};
