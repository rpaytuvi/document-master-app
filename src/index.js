import React, {Component} from 'react';
import {Provider} from 'react-redux';
import {SafeAreaProvider} from 'react-native-safe-area-context';
import {Root} from './screens';
import store from './redux';

class App extends Component {
  render = () => {
    return (
      <SafeAreaProvider>
        <Provider store={store}>
          <Root />
        </Provider>
      </SafeAreaProvider>
    );
  };
}

export default App;
