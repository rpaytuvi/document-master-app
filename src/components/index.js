import {HeaderButton, MainButton} from './buttons';
import {
  DocumentLargeListElement,
  DocumentSmallListElement,
} from './list-elements';
import {DocumentsListHeader} from './list-headers';
import {DocumentsList} from './lists';
import Footer from './footer';
import Header from './header';

export {
  HeaderButton,
  MainButton,
  DocumentLargeListElement,
  DocumentSmallListElement,
  DocumentsListHeader,
  DocumentsList,
  Footer,
  Header,
};
