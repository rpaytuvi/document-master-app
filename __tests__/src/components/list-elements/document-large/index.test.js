import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {DocumentLargeListElement} from '../../../../../src/components';

test('renders correctly', () => {
  const tree = renderer.create(<DocumentLargeListElement />).toJSON();
  expect(tree).toMatchSnapshot();
});
