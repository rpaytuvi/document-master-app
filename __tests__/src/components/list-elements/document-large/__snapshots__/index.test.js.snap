// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 4,
      "elevation": 12,
      "flexDirection": "column",
      "justifyContent": "center",
      "marginBottom": 16,
      "marginHorizontal": 16,
      "padding": 16,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.25,
      "shadowRadius": 4,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "flex-end",
        "flexDirection": "row",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#182333",
          "fontSize": 20,
          "fontWeight": "600",
          "marginRight": 8,
          "maxWidth": 200,
        }
      }
    >
      
    </Text>
    <Text
      numberOfLines={1}
      style={
        Object {
          "color": "#6f7583",
          "fontSize": 16,
        }
      }
    >
      Version 
      
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 16,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
              "marginBottom": 4,
            },
          ]
        }
      >
        <Icon
          name="groups"
          size={28}
          style={
            Object {
              "color": "#646d7a",
              "marginRight": 8,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#182333",
                "fontSize": 20,
                "fontWeight": "600",
                "marginRight": 8,
                "maxWidth": 200,
              },
              Object {
                "fontSize": 16,
              },
            ]
          }
        >
          Contributors
        </Text>
      </View>
      <View />
    </View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "flex-end",
              "flexDirection": "row",
            },
            Object {
              "alignItems": "center",
              "marginBottom": 4,
            },
          ]
        }
      >
        <Icon
          name="attachment"
          size={28}
          style={
            Object {
              "color": "#646d7a",
              "marginRight": 8,
            }
          }
        />
        <Text
          style={
            Array [
              Object {
                "color": "#182333",
                "fontSize": 20,
                "fontWeight": "600",
                "marginRight": 8,
                "maxWidth": 200,
              },
              Object {
                "fontSize": 16,
              },
            ]
          }
        >
          Attachments
        </Text>
      </View>
      <View />
    </View>
  </View>
</View>
`;
